tests/bdds/app-cleanup_test.go

```diff
@@ -0,0 +1,103 @@
+package appcleanup
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	scheme "k8s.io/client-go/kubernetes/scheme"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/litmuschaos/chaos-operator/pkg/apis/litmuschaos/v1alpha1"
+	chaosClient "github.com/litmuschaos/chaos-operator/pkg/client/clientset/versioned/typed/litmuschaos/v1alpha1"
+	restclient "k8s.io/client-go/rest"
+)
+
+var (
+	kubeconfig string
+	config     *restclient.Config
+	client     *kubernetes.Clientset
+	clientSet  *chaosClient.LitmuschaosV1alpha1Client
+)
+
+func TestChaos(t *testing.T) {
+
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "BDD test")
+}
+
+var _ = BeforeSuite(func() {
+
+	var err error
+	kubeconfig = os.Getenv("HOME") + "/.kube/config"
+	config, err = clientcmd.BuildConfigFromFlags("", kubeconfig)
+
+	if err != nil {
+		Expect(err).To(BeNil(), "failed to get config")
+	}
+
+	client, err = kubernetes.NewForConfig(config)
+
+	if err != nil {
+		Expect(err).To(BeNil(), "failed to get client")
+	}
+
+	clientSet, err = chaosClient.NewForConfig(config)
+
+	if err != nil {
+		Expect(err).To(BeNil(), "failed to get clientSet")
+	}
+
+	err = v1alpha1.AddToScheme(scheme.Scheme)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+})
+
+//BDD Tests to check secondary resources
+var _ = Describe("BDD of Application Cleanup", func() {
+
+	// BDD TEST CASE 1
+	Context("Check for the application", func() {
+
+		It("Should check for deletion of application,pv and service", func() {
+
+			//Removing Application
+			var err error
+			By("Deleting Application,Service and PVC")
+			err = exec.Command("kubectl", "delete", "-f", "../../percona/deployment.yml").Run()
+			Expect(err).To(BeNil(), "failed to delete application and its components")
+			if err != nil {
+				fmt.Println(err)
+			}
+
+			//Get the status of percona Application
+			app, err := client.AppsV1().Deployments("litmus").Get("percona", metav1.GetOptions{})
+			if err == nil {
+				count := 0
+				for app.Status.AvailableReplicas != 0 {
+					if count < 50 {
+						fmt.Printf("Percona Application is Removing the current available count is: %v \n", app.Status.AvailableReplicas)
+						app, _ = client.AppsV1().Deployments("litmus").Get("percona", metav1.GetOptions{})
+						time.Sleep(10 * time.Second)
```

Comment:
instead of random sleep use some logic to check the status

Reply:
In the next PR you can do this create an issue for the same

Reply:
But @chandankumar4 I'm not using sleep for random here. It is in the for loop, it is sleeping for 10 sec then checking the status if the status is not `Running` it is again sleeping for 10 sec.

Reply:
okay! Then it's fine